homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Config flow for Trello integration."""
+from typing import Any
+
+from trello import Member, TrelloClient, Unauthorized
+import voluptuous as vol
+from voluptuous.schema_builder import Schema
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from . import TrelloAdapter
+from .const import (
+    CONF_API_TOKEN,
+    CONF_OPTIONS_BOARDS,
+    CONF_USER_EMAIL,
+    CONF_USER_ID,
+    DOMAIN,
+    LOGGER,
+    USER_INPUT_BOARD_IDS,
+)
+
+CREDS_FORM_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_API_TOKEN): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the initial setup of a Trello integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.api_key: str = ""
+        self.api_token: str = ""
+        self.user_email: str = ""
+        self.user_id: str = ""
+        self.ids_boards: dict[str, dict[str, str]] = {}
+        self.trello_adapter: TrelloAdapter
```

Comment:
Putting everything in a dict solves the problem of having to initialize every value

Reply:
The big drawback of using a `dict` here is that you can't plainly see what information the config flow is gathering to use to create the config entry.

This could be cleaned up with a data object, but the values would still need to be initialized. In it's current state, I think there are few enough members where that abstraction isn't necessary quite yet, though.